build-in-container/run.sh

```diff
@@ -0,0 +1,52 @@
+#! /bin/bash
+
+help() {
+    echo "------------ Mariner Build-in-Container ------------"
+    echo "
+    The run.sh script presents these options
+    -t      creates container image
+    -b      creates container, 
+            builds the specs under $(pwd)/SPECS/ 
+            and places the output under $(pwd)/out/
+    -i      create interactive Mariner build container
+    -c      cleans up the current workspace, container images and instances
+    --help  shows help
+    "
+    echo "----------------------------------------------------"
+}
+
+create_container() {
+    echo "Creating Container Image"
+    source create-build-container.sh
+}
+
+build_mariner() {
+    echo "Creating Mariner Build Container and building Mariner SPECS"
+    source mariner-docker-run.sh build
+}
+
+interactive_container() {
+    echo "Creating Interactive Mariner Build Container"
+    source mariner-docker-run.sh interactive
+}
+
+cleanup() {
+    echo "Cleaning up ....."
+    rm -rf build ccache out scripts/toolkit scripts/out
+    # remove Mariner dockers
+    docker rm -f $(docker ps -a | grep mariner |tr -s " "| cut -d ' ' -f 1)
+    # remove Mariner docker images
+    docker rmi -f $(docker image ls | grep mariner |tr -s " "| cut -d ' ' -f 3)
+}
+
+while (( "$#")); do
+  case "$1" in
+    -t ) create_container; exit 0 ;;
+    -b ) build_mariner; exit 0 ;;
+    -i ) interactive_container; exit 0 ;;
```

Comment:
Interactive mode should be a major focus for us. Customers will need to pass a lot of additional flags to the tools for custom repos etc. Piping all of those extra flags through will be a lot of effort that is likely wasted.

We should very clearly document the interactive flow (do we need to set any special paths or variables to make the chroots work, etc.?)

If so, we should export those into the env maybe?

Reply:
Does this sound good for the interactive mode:
the container has a script that sets up the basic build structure and build env variables, and lands the user into toolkit. Then the user can run whatever mariner make commands they want

Reply:
Should we make a distinction of which scripts can be run within the container versus outside? Right now, the scripts under <workspace-root>/scripts can be executed inside the container.